### PR TITLE
[CI] Add ruby version for tests; format structure

### PR DESCRIPTION
**Pull request by tenhishadow**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
remove useless code
add ruby matrix to test in xcode

### Description
Just adding yaml aliases for better code management as in https://github.com/fastlane/fastlane/pull/15833

### Testing Steps
tenhi@vl-dev01:.circleci[test]$ circleci config validate config.yml 
Config file at config.yml is valid.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -76,7 +76,7 @@ jobs:
           command: |
             mkdir -p ~/test-reports
             echo ${_RUBY_VERSION} > .ruby-version
-            brew untap caskroom/versions
+            brew untap caskroom/versions || true
```
Comment (tenhishadow) [inline .circleci/config.yml]: macos users: is it okay to fix it so?
I han an issue in ci first time:
https://circleci.com/gh/fastlane/fastlane/40777

**Comment by janpio** [comment]
Omg omg, how does this even work!? "yaml alias", never heard of that before <3

Can we get the "nicer" names back somehow? Then I have absolutely no objection here.

**Comment by tenhishadow** [comment]
@janpio it's very easy :)

those ugly names like this?
"Execute tests on macOS (Xcode 9.0.1, Ruby 2.3)":

**Comment by janpio** [comment]
Haha, yes - those "ugly" names 🗡

**Comment by tenhishadow** [comment]
@janpio - ok, done

**Review by janpio — APPROVED** [review]

**Comment by tenhishadow** [comment]
@janpio could you please merge it? You have no access to do it

File: .circleci/config.yml [hunk 2]
```diff
@@ -56,138 +56,99 @@ aliases:
...
+      _RUBY_VERSION: '2.5'
+  'Execute tests on macOS (Xcode 11.3.0, Ruby 2.6)':
+    <<: *tests_macos
+    macos:
+      xcode: '11.3.0'
+    environment:
+      _RUBY_VERSION: '2.6'
 
-  "Execute tests on Ubuntu":
+  'tests_ubuntu':
```
Comment (janpio) [inline .circleci/config.yml]: Ah, just noticed: This should also be turned back into `Execute tests on Ubuntu`.

**Comment by janpio** [comment]
I will want one more review from one of my colleagues.

File: .circleci/config.yml (hunk 2 above)
Comment (tenhishadow) [reply .circleci/config.yml]: done

File: .circleci/config.yml (hunk 1 above)
Comment (joshdholtz) [reply .circleci/config.yml]: Seems fine to me 🙃

**Review by joshdholtz — APPROVED** [review]
This is beautiful 🤯 Restarted tests since it looked like one failed but I had no idea this could be done 😊 Thanks for doing this! ❤️

**Pull request merged**